From 6bdfa9f8003a5c474c6cbac833836b05a4f40a14 Mon Sep 17 00:00:00 2001
From: Michel Jauffres <michelx.jauffres@intel.com>
Date: Tue, 31 Jul 2012 14:15:06 +0200
Subject: Integrate logcat for recovery OS in Jelly Bean

BZ: 22489

[BOOT CAMP FEATURE: PUPDR]

Patch consists in adding in Android.mk the build of logcat_static like what is already
done in ICS.

Target path is modified to avoid installation of logcat_static under ./system/ directory.

This logcat_statics will be copied by Recovery_Ota.mk to sbin/logcat
(copy was made by build/core/Makefile in ICS.

Category: aosp improvement
Domain: PUPDR-Recovery
Origin: internal
Upstream-Candidate: no, need rework

Change-Id: I42dddae80ebe066a79432eacd2b1069930f44a32
Orig-Change-Id: Id75445bf2e4802d7a2bf042fb3e077f0223abb0e
Signed-off-by: Michel Jauffres <michelx.jauffres@intel.com>
Reviewed-on: http://android.intel.com:8080/59786
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Romieu, Benoit <benoit.romieu@intel.com>
Reviewed-by: Pince, Jean-christophe <jean-christophe.pince@intel.com>
Reviewed-by: Imberton, Guilhem <guilhem.imberton@intel.com>
Tested-by: Romieu, Benoit <benoit.romieu@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/74490
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 logcat/Android.mk |   17 +++++++++++++++++
 1 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/logcat/Android.mk b/logcat/Android.mk
index 7b8eb01..f0a5b2d 100644
--- a/logcat/Android.mk
+++ b/logcat/Android.mk
@@ -10,3 +10,20 @@ LOCAL_SHARED_LIBRARIES := liblog
 LOCAL_MODULE:= logcat
 
 include $(BUILD_EXECUTABLE)
+
+include $(CLEAR_VARS)
+
+LOCAL_MODULE_TAGS := optional
+
+LOCAL_MODULE:= logcat_static
+
+LOCAL_FORCE_STATIC_EXECUTABLE := true
+
+# Set path to avoid logcat_static installed under system directory
+LOCAL_MODULE_PATH := $(TARGET_RECOVERY_ROOT_OUT)/sbin
+
+LOCAL_SRC_FILES:= logcat.cpp event.logtags
+
+LOCAL_STATIC_LIBRARIES := liblog libc libstdc++
+
+include $(BUILD_EXECUTABLE)
-- 
1.7.4.1

